### PR TITLE
added winforms open_file_dialog testcase and fixed #948

### LINKED ISSUE #948: Open_File_Dialog crashes on W10 with custom file_type
following code:
```
fname = self.main_window.open_file_dialog(
  title="Open RTF file",
  multiselect=False,
  file_types=['rtf']
)
```

## Expected Behavior
should work fine ond every platform

## Current Behavior
works fine on linux, crashes on Windows 10:
```
Traceback (most recent call last):
  File "C:\Users\user\AppData\Local\Programs\NewGAN Manager\\app\newganmanager\app.py", line 301, in action_open_file_dialog
    file_types=['rtf']
  File "C:\Users\user\AppData\Local\Programs\NewGAN Manager\\app_packages\toga\window.py", line 282, in open_file_dialog
    return self._impl.open_file_dialog(title, initial_directory, file_types, multiselect)
  File "C:\Users\user\AppData\Local\Programs\NewGAN Manager\\app_packages\toga_winforms\window.py", line 182, in open_file_dialog
    "|All files (*.*)|*.*"
System.ArgumentException: Die angegebene Filterzeichenfolge ist ungültig. Sie muss eine Beschreibung des Filters gefolgt von einem vertikalen Strich (|) und dem Filtermuster enthalten. Die Zeichenfolgen für andere Filteroptionen müssen ebenfalls durch einen vertikalen Strich getrennt werden. Beispiel: "Textdateien (*.txt)|*.txt|Alle Dateien (*.*)|*.*".
   bei System.Windows.Forms.FileDialog.set_Filter(String value)
```
Translated: The provided filter-string is invalid. It must consist of a description of the filter a vertical bar (|) and the filter-pattern. Following the example of the FIXME

## Steps to reproduce
see snippet.

* Python Version (list the specific version number)
python 3.7

* Operating System and Version (select from the following and list the specific version number; if your OS is not listed, list that as well)

  - [x] Windows - version: 10

* Toga Version (list the specific version number or git hash)

* Toga Target (the type of app you are trying to generate)

  - [ ] android
  - [ ] cocoa
  - [ ] django
  - [x] gtk
  - [ ] iOS
  - [ ] tvOS
  - [ ] watchOS
  - [x] winforms
  - [ ] win32
  - [ ] Other (please specify)

## Proposed Solution
change https://github.com/beeware/toga/blob/master/src/winforms/toga_winforms/window.py#L181
to
```
file_string = "{0} files (*.{0})|*.{0}"
fil = '|'.join([file_string.format(ext) for ext in file_types]) + \
                            "|All files (*.*)|*.*"

```
This is ofc not an ideal solution, because we just add a generic description, but since there is no wrapper for an appropriate datatype I feel this would be a reasonable bridging solution.

src: https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.filedialog.filter?view=netcore-3.1

P.S: I can't contribute to windows myself, because I barely get my hands on a W10 machine and I just had a few minutes to test it out. Cheers

### PR DESCRIPTION
**Pull request by Maradonna90**
Fixs #948 
changed the file_type string generation for winforms file_dialog. Provided a test that catches the undiscovered error. Test needs more refinement (e.g. does not cancel the wait form winforms).  Some guidance would be nice.

I think creating a good test for this is the real challenge here. I might need some advice here. I used Python3.7, maybe with Python3.8 `awaited_once` unittest function this can be solved more elegant?
## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/winforms/tests/test_implementation.py [hunk 1]
```diff
@@ -11,3 +11,24 @@
...
 )
+
+from unittest.mock import patch, MagicMock
+import toga
+import toga_winforms
+import toga_dummy
+from toga.command import CommandSet
+from toga_dummy.utils import TestCase
+
+class TestWindow(TestCase):
```
Comment (freakboy3742) [inline src/winforms/tests/test_implementation.py:22]: This test should be in a separate test file (probably ``test_window.py``), not appended to the end of test_implementations.

**Comment by Maradonna90** [comment]
I am done so far. A basic test is created with `test_window.py`. The filter construction got a new method `build_filter(filter_types)`.

**Comment by Maradonna90** [comment]
Anything still missing, or is it ok as is?

**Comment by freakboy3742** [comment]
My apologies - this one fell off my radar. I'll take a look ASAP.

**Pull request merged**